packages/web-components/tests/e2e-storybook/cypress/integration/carousel/carousel.e2e.js

```diff
@@ -0,0 +1,343 @@
+          expect($duration).not.to.be.empty;
+        });
+    });
+  },
+  checkSameHeight: () => {
+    it('should have headings all the same height', () => {
+      cy.get(_selectors.heading)
+        .wait(1000)
+        .then($heading => {
+          let headingHeightNum = $heading.first().height();
+          let headingHeight = headingHeightNum + 'px';
+
+          expect($heading).to.not.have.css('height', '0px');
+          expect($heading).to.have.css('height', headingHeight);
+        });
+    });
+  },
+  checkClickableCard: () => {
+    it("should check that the footer's pseudo class takes up entire card to be clickable", () => {
+      cy.get(_selectorBase).then($carousel => {
+        if ($carousel.find(_selectors.footer).length > 0) {
+          cy.get(_selectors.footer)
+            .shadow()
+            .find('a.bx--card__footer')
+            .then($link => {
+              const url = $link.prop('href');
+              expect(url).not.to.be.empty;
+
+              const win = $link[0].ownerDocument.defaultView;
+              const after = win.getComputedStyle($link[0], ':after');
+              const positionValue = after.getPropertyValue('position');
+              const insetValue = after.getPropertyValue('inset');
+
+              expect(positionValue).to.eq('absolute');
+              if (Cypress.browser.name !== 'firefox') {
+                expect(insetValue).to.eq('0px');
+              }
+            });
+        }
+
+        if ($carousel.find(_selectors.videoFooter).length > 0) {
+          cy.get(_selectors.videoFooter)
+            .shadow()
+            .find('a.bx--card__footer')
+            .then($link => {
+              const url = $link.prop('href');
+              expect(url).not.to.be.empty;
+
+              const win = $link[0].ownerDocument.defaultView;
+              const after = win.getComputedStyle($link[0], ':after');
+              const positionValue = after.getPropertyValue('position');
+              const insetValue = after.getPropertyValue('inset');
+
+              expect(positionValue).to.eq('absolute');
+              if (Cypress.browser.name !== 'firefox') {
+                expect(insetValue).to.eq('0px');
+              }
+            });
+        }
+      });
+    });
+  },
+  checkScroll: () => {
+    it('should scroll forward when Next button is clicked and back when the Previous button is clicked', () => {
+      cy.get(_selectors.buttonNext).click();
+
+      cy.takeSnapshots();
+
+      cy.get(_selectors.buttonPrevious).click();
+
+      cy.takeSnapshots();
+    });
+  },
+};
+
+describe('dds-carousel | default (desktop)', () => {
+  beforeEach(() => {
+    cy.viewport(1280, 720);
+    cy.visit(`${_paths.default}`);
+    cy.injectAxe();
+  });
+
+  it('should check a11y', _tests.checkA11y);
+  it('should render correctly in all themes', _tests.screenshotThemes);
+
+  _tests.checkTextRenders();
+  _tests.checkSameHeight();
+  _tests.checkClickableCard();
+  _tests.checkScroll();
```

Comment:
You're sort of mixing styles here, I think we probably want all the `it('does a thing', function())` syntax to be down here inside the `describe` function